lib/everypolitician/popolo/collection.rb

```diff
@@ -31,8 +31,14 @@ def find_by(attributes = {})
       end
 
       def where(attributes = {})
-        select do |object|
-          attributes.all? { |k, v| object.send(k) == v }
+        index_attributes(attributes)
+        attributes.keys.map { |k| @indexes[k.to_sym][attributes[k]] }.reduce(:&) || []
```

Comment:
rather that iterating over the keys like this, and then looking up the values again, it would be a little cleaner to iterate over the pairs:

```ruby
attributes.map { |k,v| @indexes[k.to_sym][v] }
```